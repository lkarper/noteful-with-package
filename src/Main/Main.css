main {
    padding: 10px;
    display: flex;
    width: 80%;
}

main ul {
    list-style: none;
    padding-left: 5px;
}

@media screen and (min-width: 750px) {
    main ul {
        padding-left: 25px;
    }
}