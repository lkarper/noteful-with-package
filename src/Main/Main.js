import React from 'react';
import { Route } from 'react-router-dom';
import NotesList from '../NotesList/NotesList';
import Folder from '../Folder/Folder';
import Note from '../Note/Note';
import AddFolder from '../AddFolder/AddFolder';
import AddNote from '../AddNote/AddNote';
import UpdateNote from '../UpdateNote/UpdateNote';
import NotesError from '../NotesError/NotesError';
import AddFolderError from '../AddFolderError/AddFolderError';
import AddNoteError from '../AddNoteError/AddNoteError';
import './Main.css';
import UpdateFolder from '../UpdateFolder/UpdateFolder';

const Main = () => {
    return (
        <main>
            <NotesError>
                <Route 
                    exact path="/"
                    component={NotesList}
                />
                <Route 
                    path="/note/:noteId"
                    component={Note}
                />
                <Route
                    path="/folder/:folderId"
                    component={Folder}
                />
            </NotesError>
            <AddFolderError>
                <Route
                    path="/add-folder"
                    component={AddFolder}
                />
                <Route 
                    path="/update-folder/:folderId"
                    component={UpdateFolder}
                />
            </AddFolderError>
            <AddNoteError>
                <Route
                    path="/add-note"
                    component={AddNote}
                />
                <Route 
                    path="/update-note/:noteId"
                    component={UpdateNote}
                />
            </AddNoteError>
        </main>
    );
}

export default Main;