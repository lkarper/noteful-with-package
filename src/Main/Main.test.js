import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Main from './Main';

describe('Main component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Main />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Main />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
