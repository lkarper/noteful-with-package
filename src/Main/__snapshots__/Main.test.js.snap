// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Main>
      <main>
        <NotesError>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          >
            <NotesList
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <section
                className="notes"
              >
                <h2>
                  Loading...
                </h2>
              </section>
            </NotesList>
          </Route>
          <Route
            component={[Function]}
            path="/note/:noteId"
          />
          <Route
            component={[Function]}
            path="/folder/:folderId"
          />
        </NotesError>
        <AddFolderError>
          <Route
            component={[Function]}
            path="/add-folder"
          />
          <Route
            component={[Function]}
            path="/update-folder/:folderId"
          />
        </AddFolderError>
        <AddNoteError>
          <Route
            component={[Function]}
            path="/add-note"
          />
          <Route
            component={[Function]}
            path="/update-note/:noteId"
          />
        </AddNoteError>
      </main>
    </Main>
  </Router>
</BrowserRouter>
`;
